### PR TITLE
Solaris Librados Port

### PR DESCRIPTION
**Pull request by rohanmars**
Signed-off-by: Rohan Mars <code@rohanmars.com>

### CODE REVIEW

**Comment by rohanmars** [comment]
@liewegas is https://github.com/rohanmars/ceph/commit/f6d0d414607bcb86aa285dff639527467b28674f what you where thinking for the error handling? This is the minimal set of changes for librados. I'm checking the other message headers. I've moved to this pull request, my fingers typed ":" rather then "+" too fast. I've split apart the changes.

**Comment by yehudasa** [comment]
@rohanmars all commits are missing Signed-off-by tag

**Comment by liewegas** [comment]
very close!  just a few cosmetic items, and the Signed-off-by tags.  Thanks!

File: src/msg/simple/Pipe.cc [hunk 1]
```diff
@@ -2125,16 +2200,27 @@ int Pipe::do_sendmsg(struct msghdr *msg, int len, bool more)
       assert(l == len);
     }
 
-    int r = ::sendmsg(sd, msg, MSG_NOSIGNAL | (more ? MSG_MORE : 0));
+    int r;
+#if defined(MSG_NOSIGNAL)
+    r = ::sendmsg(sd, msg, MSG_NOSIGNAL | (more ? MSG_MORE : 0));
```
Comment (yuyuyu101) [inline src/msg/simple/Pipe.cc]: @liewegas do we really need nosignal? could we handle signal interrupt at Messenger?

Comment (liewegas) [inline src/msg/simple/Pipe.cc]: 3b34e2f36358bd93a16d38cbf9e5a464d138a79b

Hrm, hard to say what that was breaking before.  Maybe it's not a problem..

**Comment by dachary** [comment]
@rohanmars please ignore the make check bot false negative ( http://tracker.ceph.com/issues/13592 )

**Comment by liewegas** [comment]
looks good to me!  let's make sure we didn't break anything.  :)

**Pull request merged**